tests/test_nn.py

```diff
@@ -7,7 +7,7 @@
 @pytest.mark.parametrize("Q", [3, 4])
 @pytest.mark.parametrize("T", [8, 16])
 @pytest.mark.parametrize("padding_mode", ["symmetric", "zeros"])
-def test_shape(J, Q, T, padding_mode):
+def test_direct_shape(J, Q, T, padding_mode):
     B, C, N = 2, 3, 2**(J+4)
```

Comment:
here i'd be curious to see a non-power of two shape for the unit test

Reply:
you're right, `DTCWT` supports both even and odd input length now